src/engine/guidance/post_processing.cpp

```diff
@@ -414,6 +414,14 @@ void collapseTurnAt(std::vector<RouteStep> &steps,
                      current_step.name_id == steps[two_back_index].name_id)
             {
                 steps[one_back_index].maneuver.instruction.type = TurnType::Continue;
+                // If we Merge onto the same street, we end up with a u-turn in some cases
+                if (bearingsAreReversed(util::bearing::reverseBearing(
+                                            one_back_step.intersections.front()
```

Comment:
The assignments target two different parts. The `type` is always changed, the `direction_modifier` only in case of the `if`